### PR TITLE
SpaceHub - Rockets, Missions and MyProfile

### PR DESCRIPTION
**Pull request by Anand8317**
Hey CR, these are the changes done in this PR by @Salwa99  @Anand8317

Here is the live demo 🚀 : ***[SpaceHub](https://spacehub.onrender.com/)***

- [x] There is a total of 3 pages, Rockets, Missions, and My profile. Users can navigate through these pages. A browser router is used for this routing.
- [x] Redux is used to manage a centralized state in the application. 
- [x] In the Rocket page, on initial rendering, a list of rockets is fetched from the API, and the redux state is updated. 
- [x] Users can reserve the rockets and make cancellations as well. These are done using dispatching the action creator and updating the store.
- [x] In the Mission page, on initial rendering, a list of missions is fetched from the API, and the redux state is updated.
- [x] The user can join the mission and opt out of it as well. These are done using dispatching the action creator and updating the store.
- [x] On the My profile page, all the rockets and missions reserved by the user are displayed.

Cheers! 🍷

### CODE REVIEW

File: src/__test__/Rockets.test.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import { Provider } from 'react-redux';
+import RocketsList from '../components/Rockets';
+import store from '../redux/configureStore';
+
+it('rockets list renders properly', () => {
+  const tree = renderer
+    .create(<Provider store={store}><RocketsList /></Provider>)
+    .toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
Comment (TuyishimireEric) [inline src/__test__/Rockets.test.js]: - [x] **You have done a great job so far** However, while running the testing, there is an error ```src/__test__/Rockets.test.js``` that needs to be fixed to complete this project, 😎😎
    -  The error messages indicate that the component **RocketsList** is expecting some props that are not being passed in. The specific props referred to are title, description, image, and id. The error message also mentions that these props are marked as required in the RocketsList component, meaning their values cannot be undefined. 

     image
 
### KEEP IT UP 👍 
_Remember, tests ensure that your code works as expected, even when you make changes to it. This helps prevent bugs and reduces the likelihood of regressions._

**Review by TuyishimireEric — CHANGES_REQUESTED** [review]
# Status: REQUIRED CHANGES :recycle: 
### Hi  @Team 👋 👋 
#### I'm @TuyishimireEric, Your code reviewer,
Good job so far! 🥇 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights 
✔️ Professional PR. 👍
✔️ No linter errors👍
✔️ Professional README👍
✔️ Clean codes 👍

## Required Changes ♻️

- [ ] **Check the comments below 👇**

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/__test__/Rockets.test.js (hunk 1 above)
Comment (Salwa99) [reply src/__test__/Rockets.test.js]: Thank you @TuyishimireEric , I'll fix it

Comment (Salwa99) [reply src/__test__/Rockets.test.js]: Screenshot from 2023-02-09 15-20-09

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @Anand8317 and @Salwa99,

Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

thumbs-up-agree


## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**